src/mixtures/mixturemodel.jl

```diff
@@ -318,7 +318,7 @@ function _mixlogpdf!(r::AbstractArray, d::AbstractMixtureModel, x)
     p = probs(d)
     n = length(r)
     Lp = Matrix{eltype(p)}(undef, n, K)
-    m = fill(-Inf, n)
+    m = similar(r)
```

Comment:
I think this will cause incorrect results, the algorithm below is based on the fact that `m` is filled with `-Inf`.

Reply:
Maybe use
```suggestion
    m = similar(r)
    fill!(m, -Inf)
```

Reply:
But `r` is not required to be a vector so we should at least do `m = similar(r, n)`. I'm not sure it really makes sense to use `similar` here though. Are we sure we don't want a `Float64` eltype?